### PR TITLE
Added offset,length, packing and swap endianess options

### PR DESCRIPTION
**Pull request by Krzysztof314**
Added option to start reading with an offset, given amount of bytes, added option to generate 16/32 bit int in generated headers, and possibly swap endianess in data.

### CODE REVIEW

File: src/bin2header.cpp [hunk 1]
```diff
@@ -17,7 +17,7 @@
 using namespace std;
 
 
-const string version = "0.2.0";
+const string version = "0.2.1";
```
Comment (AntumDeluge) [inline src/bin2header.cpp:20]: Please don't update hardcoded version information with your changes. I will do this when I am ready to create release.

**Review by AntumDeluge — CHANGES_REQUESTED** [review]

File: src/convert.cpp [hunk 2]
```diff
@@ -38,6 +42,11 @@ void setNumberDataPerLine(const unsigned int nd) { nbData = nd; }
 
 void setShowDataContent(const bool dc) { showDataContent = dc; }
 
+void setOutputBitLength(const unsigned int bl) { if ((bl==16)||(bl==32)) outlen = bl;}
```
Comment (AntumDeluge) [inline src/convert.cpp:45]: Was this meant to support 64-bit as well? Help string says "Generate header with 16 32 or 64 bit int types"

Comment (Krzysztof314) [reply src/convert.cpp:45]: Probably I was too fast. For now I will remove 64 bits. 16 and 32bits are usefull for Audio files

**Comment by AntumDeluge** [comment]
Merged. Thank you very much.

**Pull request merged**